File: packages/instabug_private_views/lib/instabug_private_view.dart
import 'package:instabug_private_views/src/generated/instabug_private_view.api.g.dart';
import 'package:instabug_private_views/src/private_views_manager.dart';

export 'src/instabug_private_view.dart';
export 'src/instabug_sliver_private_view.dart';

class InstabugPrivateViewManager{
  static void enable() {
    final api = InstabugPrivateViewHostApi();
    api.init();
    InstabugPrivateViewFlutterApi.setup(PrivateViewsManager.I);
  }


}

File: packages/instabug_flutter/android/src/main/java/com/instabug/flutter/modules/InstabugPrivateView.java
package com.instabug.flutter.modules;

import android.os.Handler;
import android.os.Looper;

import com.instabug.flutter.generated.InstabugPrivateViewPigeon;
import com.instabug.flutter.util.privateViews.ScreenshotCaptor;
import com.instabug.library.internal.crossplatform.InternalCore;

import io.flutter.plugin.common.BinaryMessenger;

public class InstabugPrivateView implements InstabugPrivateViewPigeon.InstabugPrivateViewHostApi {
    PrivateViewManager privateViewManager;

    public static void init(BinaryMessenger messenger, PrivateViewManager privateViewManager) {
        final InstabugPrivateView api = new InstabugPrivateView(messenger, privateViewManager);
        InstabugPrivateViewPigeon.InstabugPrivateViewHostApi.setup(messenger, api);
    }

    public InstabugPrivateView(BinaryMessenger messenger, PrivateViewManager privateViewManager) {
        this.privateViewManager = privateViewManager;
        InstabugPrivateViewPigeon.InstabugPrivateViewHostApi.setup(messenger, this);
    }

    static long time = System.currentTimeMillis();

    @Override
    public void init() {
        InstabugApi.setScreenshotCaptor(new ScreenshotCaptor() {
            @Override
            public void capture(CapturingCallback listener) {

                (new Handler(Looper.getMainLooper())).postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        time = System.currentTimeMillis();
                        privateViewManager.mask(listener);

                    }
                }, 300);


            }
        }, InternalCore.INSTANCE);
    }
}
